elastic/kibana
# [RCA] Add from library to the investigation details page

Fixes https://github.com/elastic/kibana/issues/188587
Fixes https://github.com/elastic/kibana/issues/190331

https://github.com/user-attachments/assets/f72c7dd3-e069-44ea-9a31-b2a51276ddcc


## How to test

- Add following to your `kibana.dev.yml`
```
xpack.investigate.enabled: true
xpack.investigateApp.enabled: true
```
- Create an investigation using following command

```
curl --request POST \
  --url http://localhost:5601/api/observability/investigations \
  --header 'Authorization: Basic ZWxhc3RpYzpjaGFuZ2VtZQ==' \
  --header 'Content-Type: application/json' \
  --header 'elastic-api-version: 2023-10-31' \
  --header 'kbn-xsrf: oui' \
  --data '{
	"id": "my-awesome-id",
	"title": "my awesome investigation",
	"params": {
		"timeRange": { 
			"from": 1724081030849, "to": 1724081030849
		}
	},
	"origin": {
		"type": "blank"
	}
}'
```

- Go to investigation details page `http://localhost:5601/app/investigations/my-awesome-id`
- Click on the `Add observation chart button`
- Click `Import from library`
- A Flyout should open
- Select something from the list 
- The selected item should appear on the items list
- Delete the item
- The item should be deleted from the list

** At the moment there is an issue with the `visualization` type and it doesn't get rendered properly. I need to further investigate.
